### ISSUE TITLE
option to create backup on browser exit and also to restore when restarting browser

### ISSUE THREAD

**Issue by njthomas** [Drive4ik/simple-tab-groups#791]
I have auto-backups turned on, but I would like an option to save a backup when exiting Firefox.

Also an option to immediately restore this last-saved backup when starting up Firefox. 

Right now, I do both of these things manually.

**Comment by grahamperrin** [comment]
> … save a backup when …

Are you aware of the option to use a keyboard shortcut?

> immediately restore this last-saved backup when starting up Firefox.

This should be unnecessary. Please, what's the use case?

**Comment by njthomas** [comment]
Yes, I'm aware of the keyboard shortcuts. I just wanted to make the whole backupSTG-quitFirefox-startFirefox-reloadSTG process simpler by removing the first and last steps entirely. An option to automatically save STG state on exit and restore automatically on instantiation would make the whole thing seamless.

I have auto-backups turned on and running every hour, so that mostly takes care of the times when I forget to save manually, but it would be nice to have this be done automatically.

**Comment by JoeHetro** [comment]
I would also find such a feature invaluable.  I have been using Tab-Session-Manager, which I adopted after FireFox went through a major change now many years ago and my old tab manager ended support for FireFox.  Tab-Session-Manager has been very helpful over the years, although it never measured up to the old tab manager that I used to use, but that was a design limitation of the newer more secure FireFox.  As with many people, I have countless tabs open in my browser at any one time and there is a tendency for them to start open to the point that using the browser become less effective.  Being able to group tabs is extremely useful as it declutters the browser and enables me to use it more efficiently.  As with @njthomas, I would also like to automate the back-up upon closing the browser so that I am able to seamlessly restore all my tabs upon initiating a new browser session.

Thanks for a great add-on!
